From 6ca78ff17fa3e5fed4bc175b755936f7ec7218a0 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 15 Jun 2023 17:19:15 +0200
Subject: [PATCH 0355/1141] dmaengine: stm32-dma3: drop channel availability
 check

Reserved channel are no more registered in dmaengine, so they can't be
requested. No more need to check if channel is reserved in filter_fn() and
alloc_chan_resources() function.

Change-Id: Id70a8a2e8e3a26f6b6dfd6c509f375c9eb32f40b
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/313188
Tested-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
---
 drivers/dma/stm32-dma3.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/dma/stm32-dma3.c b/drivers/dma/stm32-dma3.c
index 068a3a755298..f9ebd65793c0 100644
--- a/drivers/dma/stm32-dma3.c
+++ b/drivers/dma/stm32-dma3.c
@@ -1391,10 +1391,6 @@ static int stm32_dma3_alloc_chan_resources(struct dma_chan *c)
 	u32 id = chan->id, csemcr, ccid;
 	int ret;
 
-	/* Check if chan is reserved (Secure or !CID-filtered or CID-filtered != CID1) */
-	if (ddata->chan_reserved & BIT(chan->id))
-		return -EPERM;
-
 	ret = pm_runtime_resume_and_get(ddata->dma_dev.dev);
 	if (ret < 0)
 		return ret;
@@ -1955,10 +1951,6 @@ static bool stm32_dma3_filter_fn(struct dma_chan *c, void *fn_param)
 		__func__, dma_chan_name(c),
 		conf->req_line, conf->ch_conf, conf->tr_conf, conf->tr_conf_ext);
 
-	/* Check if chan is reserved (Secure or !CID-filtered or CID-filtered != CID1) */
-	if (ddata->chan_reserved & BIT(chan->id))
-		return false;
-
 	if (!of_property_read_u32(c->device->dev->of_node, "dma-channel-mask", &mask))
 		if (!(mask & BIT(chan->id)))
 			return false;
-- 
2.39.2

